### PR TITLE
FEATURE: RAIL-290 Add jest

### PR DESCRIPTION
**Pull request by vaclavbohac**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/7/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/233/ : SUCCESS in 3m 21s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/8/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/234/ : FAILURE in 1m 33s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/10/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/236/ : SUCCESS in 44s

File: package.json [hunk 1]
```diff
@@ -45,25 +46,14 @@
     "grunt-contrib-yuidoc": "1.0.0",
     "grunt-gh-pages": "2.0.0",
     "grunt-grizzly": "0.12.0",
-    "grunt-karma": "2.0.0",
-    "grunt-mocha-test": "0.13.2",
+    "grunt-run": "^0.8.0",
```
Comment (vojty) [inline package.json]: caret

File: package.json [hunk 2]
```diff
@@ -36,7 +37,7 @@
     "expect.js": "0.3.1",
```
Comment (vojty) [inline package.json]: remove

File: package.json [hunk 3]
```diff
@@ -45,25 +46,14 @@
...
-    "karma-coverage": "1.1.1",
-    "karma-expect": "1.1.3",
-    "karma-junit-reporter": "1.2.0",
-    "karma-mocha": "1.3.0",
-    "karma-mocha-reporter": "2.2.2",
-    "karma-sinon": "1.0.5",
-    "karma-sourcemap-loader": "0.3.7",
-    "karma-webpack": "2.0.2",
     "lodash-webpack-plugin": "0.11.0",
     "mocha": "3.2.0",
```
Comment (vojty) [inline package.json]: remove

Comment (vojty) [reply package.json]: remove sinon as well

File: package.json [hunk 4]
```diff
@@ -76,7 +66,11 @@
     "fetch-cookie": "0.4.0",
     "invariant": "2.2.2",
     "isomorphic-fetch": "2.2.1",
+    "jest": "20.0.4",
```
Comment (vojty) [inline package.json]: move to devDeps

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/14/ : SUCCESS in 23s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/240/ : SUCCESS in 1m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/15/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/241/ : SUCCESS in 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/17/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/243/ : SUCCESS in 1m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/18/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/244/ : SUCCESS in 2m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/19/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/245/ : SUCCESS in 3m 14s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/31/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/246/ : SUCCESS in 1m 13s

**Pull request merged**